engines/freescape/games/dark.cpp

```diff
@@ -0,0 +1,152 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/config-manager.h"
+#include "common/file.h"
+
+#include "freescape/freescape.h"
+#include "freescape/language/8bitDetokeniser.h"
+
+namespace Freescape {
+
+DarkEngine::DarkEngine(OSystem *syst, const ADGameDescription *gd) : FreescapeEngine(syst, gd) {
+	_viewArea = Common::Rect(40, 24, 279, 124);
+	_playerHeightNumber = 1;
+	_playerHeights.push_back(16);
+	_playerHeights.push_back(48);
+
+	_playerHeight = _playerHeights[_playerHeightNumber];
+	_playerHeight = 64;
+	_playerWidth = 12;
+	_playerDepth = 32;
+}
+
+void DarkEngine::loadAssets() {
+	Common::SeekableReadStream *file = nullptr;
+	Common::String path = ConfMan.get("path");
+	Common::FSDirectory gameDir(path);
+
+	Common::File exe;
+	if (_renderMode == "ega") {
+		loadBundledImages();
+		file = gameDir.createReadStreamForMember("DSIDEE.EXE");
+
+		if (file == nullptr)
+			error("Failed to open DSIDEE.EXE");
+
+		loadFonts(file, 0xa113);
+		load8bitBinary(file, 0xa280, 16);
+	} else if (_renderMode == "cga") {
+		loadBundledImages();
+		file = gameDir.createReadStreamForMember("DSIDEC.EXE");
+
+		if (file == nullptr)
+			error("Failed to open DSIDEC.EXE");
+		load8bitBinary(file, 0x7bb0, 4); // TODO
+	} else
+		error("Invalid render mode %s for Dark Side", _renderMode.c_str());
+}
+
+void DarkEngine::gotoArea(uint16 areaID, int entranceID) {
+	debugC(1, kFreescapeDebugMove, "Jumping to area: %d, entrance: %d", areaID, entranceID);
+	if (!_gameStateBits.contains(areaID))
+		_gameStateBits[areaID] = 0;
+
+	assert(_areaMap.contains(areaID));
+	_currentArea = _areaMap[areaID];
+	_currentArea->show();
+
+	_currentAreaMessages.clear();
+	_currentAreaMessages.push_back(_currentArea->name);
+
+	int scale = _currentArea->getScale();
+	assert(scale > 0);
+
+	if (entranceID > 0 || areaID == 127) {
+		traverseEntrance(entranceID);
+	} else if (entranceID == 0) {
+		Math::Vector3d diff = _lastPosition - _position;
+		// debug("dif: %f %f %f", diff.x(), diff.y(), diff.z());
+		//  diff should be used to determinate which entrance to use
+		int newPos = -1;
+		if (abs(diff.x()) < abs(diff.z())) {
```

Comment:
please, use `ABS()` here

Reply:
Replaced `abs` by `ABS` in all the files